frontend/src/metabase/visualizations/lib/scalar_utils.ts

```diff
@@ -0,0 +1,32 @@
+import { formatValue } from "metabase/lib/formatting";
+import { OptionsType } from "metabase/lib/formatting/types";
+
+// used below to determine whether we show compact formatting
+export const COMPACT_MAX_WIDTH = 250;
+export const COMPACT_WIDTH_PER_DIGIT = 25;
+export const COMPACT_MIN_LENGTH = 6;
+
+export function compactifyValue(
+  formatOptions: OptionsType = {},
+  value: number,
+  width: number,
+) {
+  const fullScalarValue = formatValue(value, formatOptions);
```

Comment:
Nitpick: It might be worth specifying the the `compact: false` option here, since technically the caller can pass in a `formatOptions` object with `compact` equal to `true`. It's not a huge deal though since it's unlikely that someone would intentionally do that, but it could happen if copying or passing in a value from somewhere else.